src/bun.js/node/node_fs.zig

```diff
@@ -5389,9 +5395,6 @@ pub const NodeFS = struct {
                     else => @compileError("Impossible"),
                 }
             }
-            if (dirent_path_prev) |*p| {
```

Comment:
@nektro it was also leaking on errors because it never called deref if it failed in the middle of visiting directories.